pkg/oam/oam.go

```diff
@@ -0,0 +1,74 @@
+/*
+Copyright 2019 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package oam contains core OAM types.
+package oam
+
+import (
+	"context"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	runtimev1alpha1 "github.com/crossplane/crossplane-runtime/apis/core/v1alpha1"
+)
+
+// TraitKind contains the type metadata for a kind of an OAM trait resource.
```

Comment:
> I wonder if you would be interesting in removing some of the child resource logic

How does it work without that logic?

Reply:
> so that we could get some of this refactor in while that design is still in review?

Are you referring to https://github.com/crossplane/crossplane/pull/1454?

Reply:
> I think the types and interfaces here can be imported from https://github.com/crossplane/oam-kubernetes-runtime/blob/master/pkg/oam/oam.go

done